File: app/src/main/java/net/daverix/ajvm/jvm/PushConstOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class PushConstOperation<T> implements ByteCodeOperation {
    private final T value;

    public PushConstOperation(T value) {
        this.value = value;
    }

    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        currentFrame.push(value);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/PushConstOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class PushConstOperation<T>(private val value: T) : ByteCodeOperation {

    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        currentFrame.push(value)
    }
}
